kolibri/core/assets/src/api-resource.js

```diff
@@ -187,8 +190,12 @@ class Model {
     return promise;
   }
 
+  get orderedUrlParams() {
+    return this.resource.urlParams.map((key) => this.urlParams[key]);
+  }
+
   get url() {
-    return this.resource.modelUrl(this.id);
+    return this.resource.modelUrl(...this.orderedUrlParams, this.id);
```

Comment:
I don't fully understand how the logic works, but it would be great if we could remove assumptions about url ordering and instead rely on the named input patterns from Python